### PR TITLE
Basic layout for TUI

### PR DESCRIPTION
**Pull request by vrom911**
Resolves 🔜 

## Checklist:

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/summoner/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [ ] All new and existing tests pass.
- [x] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] I've used the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [x] I've added the `[ci skip]` text to the docs-only related commit's name.
- [ ] I changed code related to the generated project.
  - [ ] I created a new project using `summoner` and I checked that this specific feature is working as expected.

### CODE REVIEW

File: src/Summoner/Tui/Kit.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+data Test1 = Test1 Text Text
+    deriving (Show)
+data Test2 = Test2 Text Int
+    deriving (Show)
+
+-- | Global TUI state.
+data SummonKit = SummonKit
+    { summonKitTest1L :: [CheckBox Test1]
+    , summonKitTest2L :: [CheckBox Test2]
```
Comment (ChShersh) [inline src/Summoner/Tui/Kit.hs]: Later we can create our own lens rules to not write `L` suffix here manually :+1:

**Comment by willbasky** [comment]
Cool! And Does it tui?

File: src/Summoner/Tui/CheckBox.hs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    } deriving (Show)
+
+-- | Creates the list of the 'CheckBox'es from the list of data with the 'False'
+-- flags.
+toCheckBoxes :: [a] -> [CheckBox a]
+toCheckBoxes = map (`CheckBox` False)
+
+-- | Lens for 'checkboxFlag' field of the 'CheckBox' data type.
+flagL :: Lens' (CheckBox a) Bool
+flagL = lens checkboxFlag $ \checkbox newFlag -> checkbox { checkboxFlag = newFlag }
```
Comment (ChShersh) [inline src/Summoner/Tui/CheckBox.hs]: I guess this lens also can be generated...

File: src/Summoner/Tui.hs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    [ (E.editAttr,           V.white `Brick.on` V.black)
+    , (E.editFocusedAttr,    V.black `Brick.on` V.yellow)
+    , (invalidFormInputAttr, V.white `Brick.on` V.red)
+    , (focusedFormInputAttr, V.black `Brick.on` V.yellow)
+    ]
+
+draw :: Form SummonKit e SummonForm -> [Widget SummonForm]
+draw f = [C.vCenter $ C.hCenter form <=> C.hCenter help]
+    where
+        form = B.borderWithLabel (str "Form") $ padTop (Pad 1) (renderForm f)
```
Comment (ChShersh) [inline src/Summoner/Tui.hs]: Something wrong with the alignment here...

**Comment by vrom911** [comment]
@willbasky you can check the `summon-tui` executable to see the result, for the moment it's just a test form with a few checkboxes 🧀 
```
cabal new-exec summon-tui

or

stack exec summon-tui
```

**Comment by vrom911** [comment]
That's it now:
screenshot 2018-11-18 at 4 46 02 pm

**Pull request merged**